criu/flog.c

```diff
@@ -178,10 +200,12 @@ int flog_encode_msg(int fdout, unsigned int nargs, unsigned int mask, const char
 		 * a copy (FIXME implement rodata refs).
 		 */
 		if (mask & (1u << i)) {
-			p = memccpy(str_start, (void *)m->args[i], 0, mbuf_size - (str_start - mbuf));
-			if (p == NULL) {
-				fprintf(stderr, "No memory for string argument\n");
-				return -1;
+			if(m->args[i]){
```

Comment:
Please, explain this change.

Reply:
There are a few log messages while dumping that try to print a null string into the file. The m->args[i] became null during this point. When the memccpy() function tried to copy this argument, it caused a segmentation fault. Thus to avoid this I have placed a check to ensure that the value being copied is not NULL, i.e. if the m->args[i] is null we skip the memccpy function.